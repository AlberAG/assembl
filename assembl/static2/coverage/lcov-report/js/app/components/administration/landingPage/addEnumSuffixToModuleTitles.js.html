<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/administration/landingPage/addEnumSuffixToModuleTitles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">js/app/components/administration/landingPage</a> addEnumSuffixToModuleTitles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import countBy from 'lodash/countBy';
import get from 'lodash/get';
&nbsp;
import type LandingPageModule from '../../administration/landingPage/manageModules';
&nbsp;
export const addEnumSuffixToModuleTitles = (modules: Array&lt;LandingPageModule&gt;): Array&lt;LandingPageModule&gt; =&gt; {
  // Add a suffix to the title of the module if this module appears more than one time.
  // This suffix is the number of times this module appeared in the array.
  const moduleTypeTitles = modules.map(module =&gt; ({ title: module.moduleType.title }));
  const titleCounts = countBy(moduleTypeTitles, 'title');
  const duplicatesCurrentIndex = {};
  return modules.map((module) =&gt; {
    const { title } = module.moduleType;
    if (title &amp;&amp; titleCounts[title] &gt; 1) {
      duplicatesCurrentIndex[title] = get(duplicatesCurrentIndex, title, 0) + 1;
      return { ...module, moduleType: { ...module.moduleType, title: title &amp;&amp; `${title} ${duplicatesCurrentIndex[title]}` } };
    }
    return module;
  });
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
