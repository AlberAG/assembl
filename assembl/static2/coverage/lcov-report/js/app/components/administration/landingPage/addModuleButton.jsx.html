<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/administration/landingPage/addModuleButton.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">js/app/components/administration/landingPage</a> addModuleButton.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.78% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.78% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/9</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import React from 'react';
import { Button } from 'react-bootstrap';
import { Translate } from 'react-redux-i18n';
import { displayModal, closeModal } from '../../../utils/utilityManager';
&nbsp;
type Props = {
  createModule: Function,
  numberOfDuplicatesModules: number,
  numberOfEnabledModules: number,
  allDuplicatesAreChecked: boolean,
  buttonTitleTranslationKey: string
};
&nbsp;
const AddModuleButton = ({
  createModule,
  numberOfDuplicatesModules,
  numberOfEnabledModules,
  allDuplicatesAreChecked,
  buttonTitleTranslationKey
}: Props) =&gt; {
  const displayConfirmationModal = () =&gt; {
    const body = &lt;Translate value="administration.landingPage.manageModules.confirmationModal" /&gt;;
    const footer = [
      &lt;Button key="cancel" id="cancel-deleting-button" onClick={closeModal} className="button-cancel button-dark"&gt;
        &lt;Translate value="cancel" /&gt;
      &lt;/Button&gt;,
      &lt;Button
        key="add"
        id="confirm-add-tm-button"
        onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >          createModule(numberOfEnabledModules - 2, numberOfDuplicatesModules + 1);</span>
<span class="cstat-no" title="statement not covered" >          closeModal();</span>
        }}
        className="button-submit button-dark"
      &gt;
        &lt;Translate value="validate" /&gt;
      &lt;/Button&gt;
    ];
    const includeFooter = true;
    return displayModal(null, body, includeFooter, footer);
  };
  return (
    &lt;Button className="button-submit button-dark" onClick={displayConfirmationModal} disabled={!allDuplicatesAreChecked}&gt;
      &lt;Translate value={`administration.landingPage.manageModules.${buttonTitleTranslationKey}`} /&gt;
    &lt;/Button&gt;
  );
};
&nbsp;
export default AddModuleButton;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
