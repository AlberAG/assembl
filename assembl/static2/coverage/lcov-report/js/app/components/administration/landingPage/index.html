<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/administration/landingPage</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> js/app/components/administration/landingPage
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">74.26% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>75/101</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>24/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">48.72% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>19/39</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>72/96</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="addEnumSuffixToModuleTitles.js"><a href="addEnumSuffixToModuleTitles.js.html">addEnumSuffixToModuleTitles.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file medium" data-value="addModuleButton.jsx"><a href="addModuleButton.jsx.html">addModuleButton.jsx</a></td>
	<td data-value="77.78" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 77%;"></div><div class="cover-empty" style="width:23%;"></div></div></td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="9" class="abs medium">7/9</td>
	</tr>

<tr>
	<td class="file low" data-value="customizeHeader.jsx"><a href="customizeHeader.jsx.html">customizeHeader.jsx</a></td>
	<td data-value="46.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 46%;"></div><div class="cover-empty" style="width:54%;"></div></div></td>
	<td data-value="46.67" class="pct low">46.67%</td>
	<td data-value="15" class="abs low">7/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="14" class="abs medium">7/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="manageModules.jsx"><a href="manageModules.jsx.html">manageModules.jsx</a></td>
	<td data-value="69.77" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 69%;"></div><div class="cover-empty" style="width:31%;"></div></div></td>
	<td data-value="69.77" class="pct medium">69.77%</td>
	<td data-value="43" class="abs medium">30/43</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="46.67" class="pct low">46.67%</td>
	<td data-value="15" class="abs low">7/15</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="40" class="abs medium">28/40</td>
	</tr>

<tr>
	<td class="file high" data-value="moduleBlock.jsx"><a href="moduleBlock.jsx.html">moduleBlock.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="modulesPreview.jsx"><a href="modulesPreview.jsx.html">modulesPreview.jsx</a></td>
	<td data-value="84.62" class="pic high"><div class="chart"><div class="cover-fill" style="width: 84%;"></div><div class="cover-empty" style="width:16%;"></div></div></td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
	</tr>

<tr>
	<td class="file high" data-value="selectModulesForm.jsx"><a href="selectModulesForm.jsx.html">selectModulesForm.jsx</a></td>
	<td data-value="85.71" class="pic high"><div class="chart"><div class="cover-fill" style="width: 85%;"></div><div class="cover-empty" style="width:15%;"></div></div></td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
