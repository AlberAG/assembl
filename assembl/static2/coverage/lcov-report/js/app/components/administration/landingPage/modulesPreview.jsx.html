<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/administration/landingPage/modulesPreview.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">js/app/components/administration/landingPage</a> modulesPreview.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.62% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.62% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/13</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import * as React from 'react';
import { List, Map } from 'immutable';
import { I18n } from 'react-redux-i18n';
import ModuleBlock from './moduleBlock';
&nbsp;
type Props = {
  modules: List&lt;Map&gt;,
  moveModuleDown: Function,
  moveModuleUp: Function
};
&nbsp;
const ModulesPreview = ({ modules, moveModuleDown, moveModuleUp }: Props) =&gt; {
  if (modules.size &lt;= 0) {
    return null;
  }
&nbsp;
  const renderModuleBlock = (module) =&gt; {
    const id = module.get('id');
    const moduleTitle = module.getIn(['moduleType', 'title']);
    return (
      &lt;ModuleBlock
        key={id}
        moveDown={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >moveModuleDown(id)}</span>
        moveUp={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >moveModuleUp(id)}</span>
        required={module.getIn(['moduleType', 'required'])}
        title={moduleTitle || <span class="branch-1 cbranch-no" title="branch not covered" >I18n.t('administration.landingPage.manageModules.textAndMultimedia')}</span>
        withArrows={module.getIn(['moduleType', 'editableOrder'])}
      /&gt;
    );
  };
&nbsp;
  const header = modules.get(0);
  const footer = modules.get(modules.size - 1);
  return (
    &lt;div className="box modules-preview"&gt;
      &lt;div className="inner"&gt;
        {renderModuleBlock(header)}
        &lt;div className="other-modules"&gt;{modules.slice(1, -1).map(module =&gt; renderModuleBlock(module))}&lt;/div&gt;
        {renderModuleBlock(footer)}
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
&nbsp;
export default ModulesPreview;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
