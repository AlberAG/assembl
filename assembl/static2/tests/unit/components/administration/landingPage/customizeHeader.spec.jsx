import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { DumbCustomizeHeader } from '../../../../../js/app/components/administration/landingPage/customizeHeader';

describe('ManageModules component', () => {
  it('should render a page to manage the landing page header', () => {
    const handleTitleChange = jest.fn(() => {});
    const handleSubtitleChange = jest.fn(() => {});
    const handleButtonLabelChange = jest.fn(() => {});
    const handleImageChange = jest.fn(() => {});
    const handleLogoChange = jest.fn(() => {});
    const header = {
      title: 'title',
      subtitle: 'subtitle',
      buttonLabel: 'button label',
      headerImgMimeType: 'image/jpeg',
      headerImgUrl: 'img/headerImg.jpg',
      headerImgTitle: 'myHeaderImage',
      logoImgMimeType: 'image/jpeg',
      logoImgUrl: 'img/logoImg.jpg',
      logoImgTitle: 'myLogoImage'
    };
    const props = {
      header: header,
      handleTitleChange: handleTitleChange,
      handleSubtitleChange: handleSubtitleChange,
      handleButtonLabelChange: handleButtonLabelChange,
      handleImageChange: handleImageChange,
      handleLogoChange: handleLogoChange
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<DumbCustomizeHeader {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});