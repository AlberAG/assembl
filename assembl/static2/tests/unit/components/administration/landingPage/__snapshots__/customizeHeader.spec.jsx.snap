// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageModules component should render a page to manage the landing page header 1`] = `
<div
  className="admin-box"
>
  <SectionTitle
    annotation="* Fields are required."
    title="Header"
  />
  <div
    className="admin-content"
  >
    <div
      className="form-container"
    >
      <p
        className="admin-paragraph"
      >
        <Translate
          value="administration.landingPage.header.helper"
        />
      </p>
      <div
        className="margin-l"
      />
      <FormControlWithLabel
        componentClass={undefined}
        disabled={undefined}
        formControlProps={undefined}
        helperText=""
        helperUrl=""
        id={undefined}
        label="Debate title"
        labelAlwaysVisible={false}
        name={undefined}
        onChange={[Function]}
        required={true}
        type="text"
        validationErrors={undefined}
        value="title"
      />
      <FormControlWithLabel
        componentClass={undefined}
        disabled={undefined}
        formControlProps={undefined}
        helperText=""
        helperUrl=""
        id={undefined}
        label="Debate subtitle"
        labelAlwaysVisible={false}
        name={undefined}
        onChange={[Function]}
        required={true}
        type="rich-text"
        validationErrors={undefined}
        value="subtitle"
      />
      <FormGroup
        bsClass="form-group"
      >
        <label
          htmlFor="landing-page-img-header"
        >
          <Translate
            value="administration.landingPage.header.headerImage"
          />
        </label>
        <FileUploader
          fileOrUrl="img/headerImg.jpg"
          filename=""
          handleChange={[Function]}
          imgTitle="myHeaderImage"
          isAdminUploader={true}
          mimeType="image/jpeg"
          name="landing-page-img-header"
          onDeleteClick={[Function]}
          withPreview={true}
        />
        <div
          className="description-block"
        >
          <Translate
            value="administration.landingPage.header.headerDescription"
          />
        </div>
      </FormGroup>
      <FormControlWithLabel
        componentClass={undefined}
        disabled={undefined}
        formControlProps={undefined}
        helperText=""
        helperUrl=""
        id={undefined}
        label="Name of the referral button to the debate"
        labelAlwaysVisible={false}
        name={undefined}
        onChange={[Function]}
        required={true}
        type="text"
        validationErrors={undefined}
        value="button label"
      />
      <FormGroup
        bsClass="form-group"
      >
        <label
          htmlFor="landing-page-img-logo"
        >
          <Translate
            value="administration.landingPage.header.logoImage"
          />
        </label>
        <FileUploader
          fileOrUrl="img/logoImg.jpg"
          filename=""
          handleChange={[Function]}
          imgTitle="myLogoImage"
          isAdminUploader={true}
          mimeType="image/jpeg"
          name="landing-page-img-logo"
          onDeleteClick={[Function]}
          withPreview={true}
        />
        <div
          className="description-block"
        >
          <Translate
            value="administration.landingPage.header.logoDescription"
          />
        </div>
      </FormGroup>
    </div>
  </div>
</div>
`;
